### PR TITLE
tests: Partially revert #4806

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Partially revert the filter function added in #4806

I encountered an issue in #4832 so I have this PR to revert the filter function, see https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/4832/files#r2224181901

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: tests/apichecks/testdata/exceptions/missingfields.txt [hunk 1]
```diff
@@ -753,6 +753,7 @@
 [missing_field] crd=computenodegroups.compute.cnrm.cloud.google.com version=v1beta1: field ".spec.shareSettings.projectMap[].projectIdRef" is not set; neither 'external' nor 'name' are set
 [missing_field] crd=computenodegroups.compute.cnrm.cloud.google.com version=v1beta1: field ".spec.shareSettings.shareType" is not set in unstructured objects
 [missing_field] crd=computenodetemplates.compute.cnrm.cloud.google.com version=v1beta1: field ".spec.cpuOvercommitType" is not set in unstructured objects
+[missing_field] crd=computenodetemplates.compute.cnrm.cloud.google.com version=v1beta1: field ".spec.nodeType" is not set in unstructured objects
```
Comment (gemmahou) [inline tests/apichecks/testdata/exceptions/missingfields.txt:757]: The filter function mistakenly considered `nodeType` field as parent field and removed it. Because this field has the same prefix as below `nodeTypeFlexibility` field.

Added it back.

File: tests/apichecks/testdata/exceptions/missingfields.txt [hunk 2]
```diff
@@ -1209,7 +1210,6 @@
 [missing_field] crd=containerclusters.container.cnrm.cloud.google.com version=v1beta1: field ".spec.authenticatorGroupsConfig.securityGroup" is not set in unstructured objects
 [missing_field] crd=containerclusters.container.cnrm.cloud.google.com version=v1beta1: field ".spec.binaryAuthorization.enabled" is not set in unstructured objects
 [missing_field] crd=containerclusters.container.cnrm.cloud.google.com version=v1beta1: field ".spec.binaryAuthorization.evaluationMode" is not set in unstructured objects
-[missing_field] crd=containerclusters.container.cnrm.cloud.google.com version=v1beta1: field ".spec.clusterAutoscaling.autoProvisioningDefaults.management.upgradeOptions[]" is not set in unstructured objects
```
Comment (gemmahou) [inline tests/apichecks/testdata/exceptions/missingfields.txt:1214]: The filter function mistakenly added parent field `upgradeOptions[]`, all its children are specified in outputonlyspecfields.txt: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/252b9988877088e3f08e7f9144337f931606813d/tests/apichecks/testdata/exceptions/outputonlyspecfields.txt#L1-L2

Removed it.

File: tests/apichecks/testdata/exceptions/missingfields.txt [hunk 3]
```diff
@@ -3332,7 +3332,6 @@
 [missing_field] crd=sqlinstances.sql.cnrm.cloud.google.com version=v1beta1: field ".spec.settings.pricingPlan" is not set in unstructured objects
 [missing_field] crd=sqlinstances.sql.cnrm.cloud.google.com version=v1beta1: field ".spec.settings.replicationType" is not set in unstructured objects
 [missing_field] crd=sqlusers.sql.cnrm.cloud.google.com version=v1beta1: field ".spec.password.value" is not set in unstructured objects
-[missing_field] crd=sqlusers.sql.cnrm.cloud.google.com version=v1beta1: field ".spec.passwordPolicy.status[]" is not set in unstructured objects
```
Comment (gemmahou) [inline tests/apichecks/testdata/exceptions/missingfields.txt:3333]: The filter function mistakenly added parent field status[], all its children are specified in outputonlyspecfields.txt:
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/252b9988877088e3f08e7f9144337f931606813d/tests/apichecks/testdata/exceptions/outputonlyspecfields.txt#L4-L5

Removed it.

File: tests/apichecks/crds_test.go [hunk 4]
```diff
@@ -513,6 +513,16 @@ func TestCRDFieldPresenceInTests(t *testing.T) {
 					return
 				}
 
+				// Skip non-terminal fields (fields with children or slices)
+				if field.props != nil {
+					if len(field.props.Properties) > 0 {
```
Comment (gemmahou) [inline tests/apichecks/crds_test.go]: Previously we had `len(field.props.Properties) > 0 || field.props.Type == "object"` in the if statement. 

`object` type contains `map` so it should not be an indicator of terminal fields. If we encountered any cases related to `object` field in the future, we could handle them separately.

Comment (maqiuyujoyce) [reply tests/apichecks/crds_test.go]: When `field.props.Type == "object"`, `len(field.props.Properties) == 0` and `field.props.AdditionaProperties.type` is not `object` or `array`, then it is indeed a terminal field. Could you try it out?

Comment (maqiuyujoyce) [reply tests/apichecks/crds_test.go]: For example, https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml#L67 vs. https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_billingbudgetsbudgets.billingbudgets.cnrm.cloud.google.com.yaml#L288

Comment (gemmahou) [reply tests/apichecks/crds_test.go]: Edited: `AdditionaProperties` is `*JSONSchemaPropsOrBool` type([ref](https://github.com/kubernetes/apiextensions-apiserver/blob/v0.33.3/pkg/apis/apiextensions/types_jsonschema.go#L71)) so we could use `field.props.AdditionalProperties.**Schema**.Type`. I also tried to use Gemini CLI to refine the logic of non-terminal fields detection and it gave me similar result:
My version:
```
if field.props != nil {
	if len(field.props.Properties) > 0 {
		return
	}
	if field.props.Type == "array" && field.props.Items != nil {
		return // Skip the array itself; focus on its elements
	}
}
```
Gemini version:
```
if field.props != nil {                                                                                                                                                
  switch field.props.Type {                                                                                                                                          
  case "object":                                                                                                                                                                                                                                                  
    if len(field.props.Properties) > 0 {                                                                                                
      return                                                                                                                                                   
    }                                                                                                                                                                                                                                                    
  case "array":                                                                                                                                                                                                                                                               
   return                                                                                                                                                         
  }                                                                                                                                                                  
}                          
```
I also added detailed comment to explain the logic, so we could make improvements in the future if any edge cases occur.

Comment (maqiuyujoyce) [reply tests/apichecks/crds_test.go]: The new logic won't try to check if the fields in the additionalProperties is set in the test or not (verified by removing `values` field in the test case for BillingBudgetsBudget). Could you ensure this edge case is covered?

Comment (gemmahou) [reply tests/apichecks/crds_test.go]: This is a real edge case since even the existing logic doesn't check fields under AdditionalProperties. I've handled this in a separate PR: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/4920

**Comment by justinsb** [comment]
@gemmahou looks like you have approve - can you please rebase + make sure tests pass.
